src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1044,10 +1048,10 @@ public override WebResponse GetResponse()
 
         public override Stream GetRequestStream()
         {
-            return InternalGetRequestStream().Result;
+            return InternalGetRequestStream(async: false).Result;
         }
 
-        private Task<Stream> InternalGetRequestStream()
+        private Task<Stream> InternalGetRequestStream(bool async)
```

Comment:
> It even gets ignored in SendRequest if we don't allow buffering, so the arg is moot here.

We're ignoring it because we're keeping the Sync path in case we're buffering, but we can't keep the Sync path if we're not buffering because we should open the connection upfront (`by starting request task`) and if we try to use Sync over thread we're going to lock the whole thread.

Reply:
> Why does asynchronicity of getting the request stream affects the asynchronicity of invoking `HttpClient.Send(Async?)`? I should be able to do:
> 
> ```
> GetRequestStream() - synchronously retrieve the request content stream
> // write to stream asynchronously
> await TaskToApm(BeginGetResponse, EndGetResponse) - asynchronously invoke the request
> ```

That's a good point actually, and with the current workflow that should be possible. But as you already mentioned this argument might not be necessary here, because in case we're not buffering we're triggering the async path anyway. I will delete this parameter.